### PR TITLE
CB-13336 Allow clusters with ephemeral storage stopped if there are EBS disks attached

### PR DESCRIPTION
**Pull request by daszabo**

### CODE REVIEW

**Comment by doktoric** [comment]
please add some UT coverage

**Comment by daszabo** [comment]
> please add some UT coverage

Done

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterRepairService.java [hunk 1]
```diff
@@ -369,7 +369,7 @@ private boolean hasReattachSupportedVolumes(Stack stack, String hostGroupName) {
                 .map(InstanceGroup::getTemplate)
                 .map(Template::getVolumeTemplates)
                 .anyMatch(volumes -> volumes.stream()
-                        .map(VolumeTemplate::getVolumeType).anyMatch(REATTACH_NOT_SUPPORTED_VOLUME_TYPES::contains));
+                        .map(VolumeTemplate::getVolumeType).allMatch(REATTACH_NOT_SUPPORTED_VOLUME_TYPES::contains));
```
Comment (pdarvasi) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterRepairService.java:372]: this is confusing for me, the method name (hasReattachSupportedVolumes) contradicts the last line
```.map(VolumeTemplate::getVolumeType).allMatch(REATTACH_NOT_SUPPORTED_VOLUME_TYPES::contains));```

What does this really do?

File: core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackStopRestrictionService.java [hunk 2]
```diff
@@ -18,7 +18,7 @@ public StopRestrictionReason isInfrastructureStoppable(String cloudPlatform, Set
         if (SUPPORTED_CLOUD_PLATFORM.equals(cloudPlatform)) {
             for (InstanceGroup instanceGroup : instanceGroups) {
                 if (instanceGroup.getTemplate().getVolumeTemplates().stream()
-                        .anyMatch(volume -> AwsDiskType.Ephemeral.value().equals(volume.getVolumeType()))) {
+                        .allMatch(volume -> AwsDiskType.Ephemeral.value().equals(volume.getVolumeType()))) {
                     reason = StopRestrictionReason.EPHEMERAL_VOLUMES;
```
Comment (pdarvasi) [inline core/src/main/java/com/sequenceiq/cloudbreak/service/stack/StackStopRestrictionService.java:22]: could you pls log this event?

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/TemplateValidator.java [hunk 3]
```diff
@@ -154,7 +154,9 @@ private void validateMaximumVolumeSize(Template value, VmType vmType, Validation
         if (vmType != null) {
             Object maxSize = vmType.getMetaDataValue(VmTypeMeta.MAXIMUM_PERSISTENT_DISKS_SIZE_GB);
             if (maxSize != null) {
-                int fullSize = value.getVolumeTemplates().stream().mapToInt(volume -> volume.getVolumeSize() * volume.getVolumeCount()).sum();
+                int fullSize = value.getVolumeTemplates().stream()
+//                        .filter(volumeTemplate -> !AwsDiskType.Ephemeral.value().equals(volumeTemplate.getVolumeType()))
```
[the remark this answers is not in the document]
Comment (daszabo) [reply core/src/main/java/com/sequenceiq/cloudbreak/controller/validation/template/TemplateValidator.java]: no, this was a mistake

File: core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterRepairService.java (hunk 1 above)
Comment (daszabo) [reply core/src/main/java/com/sequenceiq/cloudbreak/service/cluster/ClusterRepairService.java:372]: Renamed

**Review by pdarvasi — APPROVED** [review]

**Pull request merged**